vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java

```diff
@@ -253,23 +236,98 @@ public I addItem(Component component) {
      *            the components to add
      * @see HasMenuItems#addItem(String, ComponentEventListener)
      * @see HasMenuItems#addItem(Component, ComponentEventListener)
+     *
+     * @deprecated Since 24.8, use {@link #addItem(Component)} instead
      */
+    @Deprecated(since = "24.8")
```

Comment:
Note that `add()` is used internally by `addSeparator()` and we probably should refactor that method:

https://github.com/vaadin/flow-components/blob/4e414bb3d978fcc5bf0776a02caded0aa25077a8/vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/SubMenu.java#L64-L66

Reply:
My bad, I missed that. We probably could consider adding `addSeparator()` also to `ContextMenu` for consistency, there is currently an integration test explicitly covering this use case:

https://github.com/vaadin/flow-components/blob/22d9c2b1f44f1abe7ff9a51aee3126c247f13890/vaadin-context-menu-flow-parent/vaadin-context-menu-flow-integration-tests/src/main/java/com/vaadin/flow/component/contextmenu/it/ContextMenuView.java#L155-L158

Reply:
Would that qualify as a feature? Maybe we can leave that for another PR, what do you think?